src/user/user.ts

```diff
@@ -124,40 +124,83 @@ export class User implements IUser {
 
   public setWalletInfo(
     id: string,
-    info: string | WalletDescriptor,
-    legacyWallet?: boolean
+    info: string | WalletDescriptor
   ) {
-    let walletInfo = this.getWalletInfo(id, legacyWallet);
+    let walletInfo = this.getWalletInfo(id);
     if (!walletInfo) {
+      const isLegacyWallet = this.isLegacyWalletID(id);
+
       // Not in legacy mode, we should automatically add the wallet account into the list
-      if (!legacyWallet && this.hasHDWallet()) {
+      if (!isLegacyWallet && this.hasHDWallet()) {
         this.setHDWalletAccount(id);
-        walletInfo = this.getWalletInfo(id, legacyWallet);
+        walletInfo = this.getWalletInfo(id);
       }
+
       if (!walletInfo) {
         throw new Error(`The requesting wallet is not available ${id}`);
       }
     }
     walletInfo.descriptor = WalletDescriptor.from(info);
   }
 
-  public getWalletInfo(id: string, legacyWallet?: boolean): WalletInfo {
+  public getWalletInfo(id: string): WalletInfo {
     let info: WalletInfo = null;
-    if (legacyWallet === true) {
-      if (this.legacyWallets) {
-        info = this.legacyWallets.filter((x) => x.key == id)[0];
+
+    // Start with HD wallet (as we should in this mode instead of legacy)
+    if (this.wallet) {
+      const derives = this.wallet.derivedWallets;
+      if (derives) {
+        for (let i = 0; i < derives.length; i++) {
```

Comment:
should we you `this.legacyWallets.find`  instead of for loop ?

Reply:
oh nice, gonna use it